backend/src/main/java/com/team4/backend/service/NotificationService.java

```diff
@@ -0,0 +1,46 @@
+package com.team4.backend.service;
+
+import com.mongodb.client.result.UpdateResult;
+import com.team4.backend.dto.NotificationDto;
+import com.team4.backend.mapping.NotificationMapper;
+import com.team4.backend.model.Notification;
+import com.team4.backend.repository.NotificationRepository;
+import com.team4.backend.repository.UserRepository;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.Sinks;
+
+@Service
+public class NotificationService {
+
+    private final NotificationRepository notificationRepository;
+
+    private final Sinks.Many<Notification> sinks;
+
+    public NotificationService(NotificationRepository notificationRepository, Sinks.Many<Notification> notificationFlux) {
+        this.notificationRepository = notificationRepository;
+        this.sinks = notificationFlux;
+    }
+
+    public Mono<Notification> createNotification(NotificationDto notification) {
+        return notificationRepository
+                .save(NotificationMapper.toEntity(notification))
```

Comment:
Je croyais qu'on faisait le mapping dans le controlleur pour retourner un Dto

Reply:
Le dto vient de l'extérieur, on le map to entity (on enlève le id puisqu'il est généré par la base de données)

Reply:
Dans le cas des notifications, ils ne seront pas créé de l'extérieur, quand on créer une notification on lui passe une notification dto (sans le id étant donné que la base de données le créer)